helix-term/src/application.rs

```diff
@@ -283,11 +288,21 @@ impl Application {
                                 (None, Some(message), Some(percentage)) => {
                                     format!("{}% {}", percentage, message)
                                 }
-                                (Some(title), None, None) => title,
-                                (None, Some(message), None) => message,
+                                (Some(title), None, None) => title.to_string(),
+                                (None, Some(message), None) => message.to_string(),
                                 (None, None, Some(percentage)) => format!("{}%", percentage),
                                 (None, None, None) => "".into(),
+                            };
+
+                            if let lsp::WorkDoneProgress::End(_) = work {
+                                self.lsp_progress.remove(&token);
                             }
+
+                            msg
+                        };
+                        let token = match token {
+                            lsp::NumberOrString::Number(n) => n.to_string(),
+                            lsp::NumberOrString::String(s) => s,
```

Comment:
> Sorry but I'm not fully sure which buffer are you talking about, could you point it out to me perhaps?

This part. `String` is being allocated and deallocated for every event.

Reply:
I think I optimized it somewhat, let me know if you think we can improve it further. Currently the `status` variable is moved to `set_editor` so we still have to allocate it on each event.